### PR TITLE
separate view and presenter for header

### PR DESCRIPTION
**Pull request by Anthial**

### CODE REVIEW

File: src/frontend/views/header.tsx [hunk 1]
```diff
@@ -30,7 +39,7 @@ export default function Header(){
           <h1 className="font-bold text-2xl py-2 mr-20">My Friday Night List</h1>
           <div className="flex">
             <div className="flex w-72 mr-4">
-              <input className="outline-none w-72 rounded-lg bg-[#212F45] pl-4" type="text" placeholder="Search..." />
+              <input className="outline-none w-72 rounded-lg bg-[#212F45] pl-4" type="text" placeholder="Search..." value={props.value} onChange={(e) => props.setQuery(e.target.value)} onKeyDown={(e)=> {if (e.key === "Enter") props.search()}}/>
```
Comment (hannesmann) [inline src/frontend/views/header.tsx:42]: The button should be disabled if the user hasn't entered a search value

```
disabled={props.value.trim() === ""}
```

**Review by hannesmann — CHANGES_REQUESTED** [review]

**Comment by adrian-jonsson-sjoedin** [comment]
text imputed in the search field persist when navigating around the site. Do we want this?  I think it would be better if the search field is cleared after search or when switching page.

File: src/frontend/views/header.tsx (hunk 1 above)
Comment (Anthial) [reply src/frontend/views/header.tsx:42]: This is not correct.

**Review by hannesmann — DISMISSED** [review]

**Review by adrian-jonsson-sjoedin — APPROVED** [review]
LGTM

**Pull request merged**